rfc-029-repositories-should-be-public-by-default.md

```diff
@@ -0,0 +1,81 @@
+  public and keeping it private for our own use or that of the client.
+- Sometimes the code that we write might be useful to the rest of the world –
+  it solves a common problem, it provides inspiration for how to solve a
+  problem
+- We don’t usually work on projects that require secrecy.
+- We already make most of our repositories public – most notably the Playbook –
+    but until now we haven’t had an official policy, although we’ve [tried
+  to](https://github.com/dxw/playbook/issues/12) in the past. Most of our client
+  projects are also in public repositories.
+- We’ve [already talked
+  about](rfc-018-our-decision-making-process-is-public.md) why we make our RFC
+  decision making process open – the same reasons apply here
+
+Allowing the team to work in the open also has benefits for individual
+developers. They:
+
+- can show their work to friends and family – show the thing instead of talking
+  about the thing
+- know that they’re investing their energy in something maximally useful and
+  long-lasting
+- can build up a public portfolio of work
+
+## Proposal
+
+All new GitHub respositories in the dxw organisation SHOULD be public, unless
+they directly relate to a client project.
+
+- TODO – I can’t think of many reasons why such a repo shouldn’t be public –
+  perhaps it contains secrets e.g. keys? Bring this up at tech team forum
+- TODO – should private repos always have a rationale attached to them
+  explaining why they’re private?
+
+When starting a new client project, we SHOULD ask the client whether we can
+make the repository public. We SHOULD explain the benefits of making it public.
+Where relevant, we can refer to existing guidance such as the [Government
+Design Principles](https://www.gov.uk/guidance/government-design-principles#make-things-open-it-makes-things-better)
+or the [GOV.UK Service Standard](https://www.gov.uk/service-manual/service-standard/point-12-make-new-source-code-open).
+
+This proposal doesn’t address:
+
+- licencing of the code – we must think about this, because having a public
+  repository without a licence means that legally [nobody can do anything with
+  it](https://choosealicense.com/no-permission/)
+- things that aren’t hosted on GitHub – mainly our GovPress stuff
+
+### Concerns
+
+When we write code for internal use, we don’t always apply the same standards
+which we would for a client project. For example, we have some repositories
+which are proofs of concept, or constrained by time. If somebody outside of dxw
+sees one of these repositories without understanding the context in which it
+was made, it might give a negative impression of the quality of our work.
+
+We SHOULD reduce this risk by pinning some repositories which we’re proud of,
+so that they appear at the top of dxw’s GitHub page.
+
+## Next steps
+
+- Start a conversation about how we licence the software in our public
+  repositories, and open an RFC
```

Comment:
MIT or Public Domain

Reply:
I think a bunch of people are going to have a bunch of opinions on this – that's why I've left it as a separate thing 😄

Reply:
![Yeah, well, that's just like, your opinion, man](https://media.giphy.com/media/26BRrSvJUa0crqw4E/giphy.gif)

Reply:
I was certainly looking forward to the licence debate. (MIT or any BSD derivative would be my choice)

Reply:
I've flip-flopped on this one and brought the licence choice into this RFC. When we talked about it as a team everyone seemed pretty chill about MIT.